paddle/parameter/Parameter.h

```diff
@@ -324,6 +324,7 @@ class Parameter {
   std::vector<std::shared_ptr<IParameterUpdaterHook>> updaterHooks_;
 
 public:
+  void setSharedCount(int cnt) { sharedCount_ = cnt; }
```

Comment:
The purpose of this PR is a fix. How about we do the successive edits in future PRs? @reyoung